### PR TITLE
Enable continuous integration builds via Github Actions

### PR DESCRIPTION
**Pull request by bjosv**
Build, test, run checkers and make sure edocs can be generated.

- Verify on latest Erlang 23, 22, 21
- Cache hex packages and Dialyzer PLTs
- Pin elvis version to 0.5.0
- Remove unused Travis CI config

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Cache Hex packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/rebar3/hex/hexpm/packages
+          key: ${{ runner.os }}-hex-${{ hashFiles('**/rebar.lock') }}
+          restore-keys: ${{ runner.os }}-hex-
```
Comment (zuiderkwast) [inline .github/workflows/ci.yml:36]: I don't think we're using any Hex packages in this repo.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          make cover
+
+      - name: Run checks
+        run: |
+          make elvis
+          make xref
+          make dialyzer
+
+      - name: Generate docs
+        run: make edoc
```
Comment (zuiderkwast) [inline .github/workflows/ci.yml:73]: Future TODO: Publish the generated docs somewhere... (as .md files inside the repo itself or on GitHub Pages)

**Review by zuiderkwast — COMMENTED** [review]
Nice!

Perhaps just delete the "help" target in the Makefile:

```Makefile

help:
	@echo "Please use 'make <target>' where <target> is one of"
	@echo "  start             starts a test redis cluster"
	@echo "  stop              stops all redis servers"
	@echo "  travis-run        starts the redis cluster and runs your tests"
	@echo "  travis-install    install redis from 'unstable' branch"
	@echo "  edoc, elvis, cover and more..."
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (zuiderkwast) [reply .github/workflows/ci.yml:36]: Sorry, we do for Poolboy. (I mixed up the review with another review I had open.)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (zuiderkwast) [reply .github/workflows/ci.yml:73]: Ah...:coffee:..., this is eredis_cluster... So, we already have the generated docs for some of the erl files under doc/*.md.

To make sure these checked-in generated docs are up to date, perhaps we can let this job check that make edoc is a no-op to those generated files?

Comment (bjosv) [reply .github/workflows/ci.yml]: Ah good, removed help, and added a build breaker for missing docs changes in commits

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+name: CI
+on:
+  push:
+  pull_request:
+    branches:
+      # Branches from forks have the form 'user:branch-name' so we only run
+      # this job on pull_request events for branches that look like fork
+      # branches. Without this we would end up running this job twice for non
+      # forked PRs, once for the push and then once for opening the PR.
+    - '**:**'
```
Comment (zuiderkwast) [inline .github/workflows/ci.yml:10]: I don't know if it matters in Yaml, but is `-'**:**'` indented the same as `branches:` here?

**Review by zuiderkwast — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bjosv) [reply .github/workflows/ci.yml:10]: Yes, look better anyway, a linter I tested thought so aswell

**Pull request merged**